docs/TrafficSplitting.md

```diff
@@ -0,0 +1,157 @@
+ - Tag targets
+ - Split traffic
+
+### Tag Targets
+
+User can tag traffic targets (revisions) to
+ - Create custom URLs
+ - Create a human-friendly reference to refer to a traffic target
+ - Create static routes and change the underlying revisions as needed
+ - Remove the custom URLs OR remove the tags of targets
+ - Remove a target from traffic block altogether
+
+
+#### Tagging commands
+
+#### 1. `kn service tag-target serviceName [RevisionName1:Tag1] [RevisionName2:Tag2]`
+
+ - Takes the first argument as service name to operate on
+ - Tags given revision with specified tag thereby creating a custom URL
+ - Does not change the traffic percent of the target
+ - Multiple `RevisionName:Tag` combination can be specified
+ - Multiple tags for same revision can be specified
+   - `kn service tag-target svc echo-v1:stable echo-v1:current`
+ - If the target doesn't exist in deployed service, it creates a target entry with 0% traffic portion
+ - For updating tags of existing traffic targets, use `kn service untag-target` followed by `kn service tag-target`, because
+    - The deployed service might have multiple tags referring to a `RevisionName`
+    - The specified `RevisionName:Tag` combination on CLI might request to have different tags for same revision
+
+#### 2. `kn service untag-target serviceName [Tag] [RevisionName:Tag] [RevisionName [--all]]`
+
+ - Takes the first argument as service name to operate on
+ - Takes single reference of a traffic target
+ - Does not change the traffic percent of the target
+ - Can be specified either via
+   -  `Tag`, removes the specified tag from traffic targets
+   -  `RevisionName`, removes the tag for target where `RevisionName` matches, errors if multiple tags for a revision exists
+   -  `RevisionName:Tag` removes `Tag` for `RevisionName`, useful for case where a single revision is tagged with multiple tags
+   - `--all` flag can be optionally specified along with `RevisionName` to remove all tags for given revision
+ - If a target has 0% traffic portion and it is untagged, target is removed from traffic block altogether
+
+
+## Split Traffic
+
+A user can
+ - Specify the traffic percent portion to route to a particular target
+ - Adjust the traffic percent of existing traffic targets
+ - Reference a traffic target either by `RevisionName` or `Tag`
+
+### Traffic splitting command
+
+#### `kn service set-traffic serviceName [revisionName1=Percent] [Tag2=Percent] [Tag3=*]`
+
+ - Takes the first argument as service name to operate on
+ - Subsequent arguments specifies the traffic targets
+ - Traffic target can be referenced either via `RevisionName` or `Tag`
+ - Traffic portion to allocate is specified after `=`
+ - Wildcard character `*` can be specified to allocate portion remaining after, summing specified portions and substracting it from 100
+ - Specified traffic allocations replace existing traffic block of deployed service
+
+#### Referencing `latestRevision:true` (`$`) target
```

Comment:
Not a huge fan of the `$` here... Maybe there is a precedence in other CLIs or reasoning. Can you elucidate this choice @navidshaikh? Thanks.

Not a huge concern if there is precedence. If not, then `*latestRevision` or `&latestRevision` could be perhaps more descriptive?

Reply:
`$` is a take on regexps where it means 'the end of a line' I guess (like in 'the end of a line of revisions') ;-). A bit made up, we could definitely use something different. However we should choose best something which does not has a meaning for a shell. E.g '*latestRevision' would expand to a filename if in the current directory is a file like 'mylatestRevision' before it's handed over to `kn`. `&` will put the process in the background if unescaped. `$` alone is safe but when combined with another character it specifies a variable. 

I'm afraid regardless of what we chose we won't find something so intuitive that it's obvious without reading the documentation.

I think `$` is ok-ish but maybe we find something better ?

Reply:
one of the suggestion Naomi mentioned during WG meeting is to use flag `--traffic-latest` which should refer the `latestRevision` field in traffic block.

We'll need an identifier for (a) traffic (b) tag operations, we can use
`--traffic-latest` and `--tag-latest` flags, for eg:
1. `--traffic-latest 2`
2. `--tag-latest current`